notebook/converted/semi-supervision_feasibility.py

```diff
@@ -0,0 +1,182 @@
+            if 'tissue:' in characteristic:
+                tissue = characteristic.split(':')[1]
+                tissue = tissue.strip().lower()
+                return tissue
+            
+    # Catch exceptions caused by a field not being present
+    except KeyError:
+        return None
+    
+    # 'refinebio_annotations' is usually a length 1 list containing a dictionary.
+    # Sometimes it's a length 0 list indicating there aren't annotations
+    except IndexError:
+        return None
+
+
+# In[5]:
+
+
+sample_metadata = metadata['samples']
+
+tissues = []
+for sample in sample_metadata:
+    tissue = get_tissue(sample_metadata, sample)
+    if tissue is not None:
+        tissues.append(tissue)
+
+
+# In[6]:
+
+
+tissue_counts = collections.Counter(tissues)
+tissue_counts.most_common()[:5]
+
+
+# In[7]:
+
+
+keys = tissue_counts.keys()
+blood_keys = []
+for key in keys:
+    if 'blood' in key or 'pbmc' in key:
+        blood_keys.append(key)
+sorted(blood_keys)
+
+
+# In[8]:
+
+
+# Keep whole blood and pbmcs, leave out samples containing a single cell type
+# Also leave out umbilical cord blood because it's not quite the same thing
+# https://pubmed.ncbi.nlm.nih.gov/12634410/
+keys_to_keep = ['blood',
+                'blood (buffy coat)',
+                'blood cells',
+                'blood monocytes',
+                'blood sample',
+                'cells from whole blood',
+                'fresh venous blood anticoagulated with 50 g/ml thrombin-inhibitor lepirudin',
+                'healthy human blood',
+                'host peripheral blood',
+                'leukemic peripheral blood',
+                'monocytes isolated from pbmc',
+                'normal peripheral blood cells',
+                'pbmc',
+                'pbmcs',
+                'peripheral blood',
+                'peripheral blood (pb)',
+                'peripheral blood mononuclear cell',
+                'peripheral blood mononuclear cell (pbmc)',
+                'peripheral blood mononuclear cells',
+                'peripheral blood mononuclear cells (pbmc)',
+                'peripheral blood mononuclear cells (pbmcs)',
+                'peripheral blood mononuclear cells (pbmcs) from healthy donors',
+                'peripheral maternal blood',
+                'peripheral whole blood',
+                'periphral blood',
+                'pheripheral blood',
+                'whole blood',
+                'whole blood (wb)',
+                'whole blood, maternal peripheral',
+                'whole venous blood'
+               ]
+
+
+# In[9]:
+
+
+blood_counts = dict((k, tissue_counts[k]) for k in keys_to_keep)
+
+
+# In[10]:
+
+
+total_samples = 0
+for key in blood_counts:
+    total_samples += blood_counts[key]
+total_samples
+
+
+# ## Count unlabeled blood cells
+# ~25k blood samples is around 3x as many samples as we have labeled. Let's find exactly how much overlap there is between these samples and our labeled samples
```

Comment:
Labels correspond to disease/healthy states, though since I'm just testing whether semi-supervised learning works here, they could correspond to anything

Yeah, sets are a more reasonable way to think about it and should run slightly faster